source/System.cpp

```diff
@@ -882,6 +882,20 @@ const vector<System::Asteroid> &System::Asteroids() const
 
 
 
+// Get a list of all unique payload outfits in this system.
+const set<const Outfit *> System::Payloads() const
+{
+	set<const Outfit *> payloads;
+	for(const auto &asteroid : asteroids)
+		if(asteroid.Type())
+			for(const auto &payload : asteroid.Type()->GetPayload())
+				payloads.insert(payload.outfit);
+
+	return payloads;
+}
```

Comment:
Recreating a new set every time this function is called is a bit of an issue, as it gets called pretty often in some conditions. It should probably be cached when the system is created or modified.

Reply:
>it gets called pretty often in some conditions.

Where? I'm not finding where PlayerInfo::HasMapped() or PlayerInfo::Map() are called in any performance critical code. Github's only returning 5 hits.

Reply:
> > it gets called pretty often in some conditions.
> 
> Where? I'm not finding where PlayerInfo::HasMapped() or PlayerInfo::Map() are called in any performance critical code. Github's only returning 5 hits.

HasMapped gets called every frame if you have an outfit with a map selected in the outfitter panel

Reply:
Moving this code to `System::UpdateSystem` should resolve this.